asmcomp/asmlink.ml

```diff
@@ -276,13 +288,15 @@ let call_linker_shared file_list output_name =
 
 let link_shared ~ppf_dump objfiles output_name =
   Profile.record_call output_name (fun () ->
-    let units_tolink = List.fold_right scan_file objfiles [] in
+    let units_tolink, objfiles_rev =
+      List.fold_right scan_file objfiles ([], [])
+    in
     List.iter
       (fun (info, file_name, crc) -> check_consistency file_name info crc)
       units_tolink;
     Clflags.ccobjs := !Clflags.ccobjs @ !lib_ccobjs;
     Clflags.all_ccopts := !lib_ccopts @ !Clflags.all_ccopts;
-    let objfiles = List.rev (List.map object_file_name objfiles) @
+    let objfiles = List.map object_file_name objfiles_rev @
```

Comment:
For the reason mentioned above, I think the `List.rev` in the original code should be preserved.

Reply:
In addition: I think this is one of the situations where it's nice to use `List.rev_map`.

Reply:
Indeed - had I not been being an idiot about folds, I’d have switched it to that here!